src/plugins/event-plugins/WebVitalsPlugin.ts

```diff
@@ -14,15 +14,28 @@ import {
 import {
     CLS_EVENT_TYPE,
     FID_EVENT_TYPE,
-    LCP_EVENT_TYPE
+    LCP_EVENT_TYPE,
+    PERFORMANCE_NAVIGATION_EVENT_TYPE,
+    PERFORMANCE_RESOURCE_EVENT_TYPE
 } from '../utils/constant';
+import { Topic } from '../../event-bus/EventBus';
+import { ParsedRumEvent } from '../../dispatch/dataplane';
+import { ResourceEvent } from '../../events/resource-event';
+import {
+    HasLatency,
+    ResourceType,
+    performanceKey
+} from '../../utils/common-utils';
 
 export const WEB_VITAL_EVENT_PLUGIN_ID = 'web-vitals';
 
 export class WebVitalsPlugin extends InternalPlugin {
     constructor() {
         super(WEB_VITAL_EVENT_PLUGIN_ID);
+        this.lcpHelper = this.lcpHelper.bind(this);
     }
+    private resourceEventIds = new Map<string, string>();
+    private navigationEventsIds: string[] = [];
```

Comment:
nit: Since we only ever use the first, this doesn't need to be an array.

Reply:
agreed, changed to a private string var